project/eScavenge/Game.py

```diff
@@ -268,7 +268,17 @@ def get_status(self, now, username):
         return 'Final Points: {}'.format(current_team.points)
 
     def get_snapshot(self, now):
-        pass
+        stringList = []
+        for username in self.__teams:
+            current_team = self.get_team(username)
+            total_time = datetime.timedelta(days=0, hours=0, minutes=0, seconds=0)
+            for t in current_team.time_log:
+                total_time += t
+            if current_team.current_landmark <= len(self.__landmarks):
+                stat_str = "Team: {}\nYou Are On Landmark {}\nTime Taken For Landmarks: {}\nTotal Points: {}\n"
+                stringList.append(stat_str.format(username, current_team.current_landmark + 1, total_time,
+                                                  current_team.points))
+        return ''.join(stringList)
```

Comment:
This is just returning a string, which is fine, but to address my comment about checking state in game you can return both an error and a string. so this line would be `return Errors.SUCCESS, ''.join(stringList)`
The error case would be `return Errors.SOMEERROR, ''` (empty string)